runtime/doc/lua.txt

```diff
@@ -969,6 +969,65 @@ vim.str_byteindex({str}, {index}, {use_utf16})           *vim.str_byteindex()*
       • {index}      (number)
       • {use_utf16}  any|nil
 
+vim.str_utf_end({str}, {index})                            *vim.str_utf_end()*
+    Returns 0 if the index is pointing at the last byte of a character, 1 if
+    index is pointing at the second last byte of a character, and so on.
+
+    Examples: >lua
+
+     -- The character 'æ' are equal to the bytes '\xc3\xa6' (using UTF-8)
+
+     -- Returns 0 because the index is pointing at the last byte of a character
+     vim.str_utf_end('æ', 2)
+
+     -- Returns 1 because the index is pointing at the second last byte of a character
+     vim.str_utf_end('æ', 1)
+<
+
+    Parameters: ~
+      • {str}    (string)
+      • {index}  (number)
+
+    Return: ~
+        (number)
+
+vim.str_utf_pos({str})                                     *vim.str_utf_pos()*
+    Gets a list of the starting byte position of each char in the given UTF-8
+    encoded string based on the UTF-8 encoding.
```

Comment:
why is "based on the UTF-8 encoding" necessary to mention? It's already implied by the previous mention of "UTF-8 encoded string" .

Is this a more correct and less confusing wording:

```suggestion
    Gets a list of the starting byte positions of each UTF-8 codepoint in the given string.
```